sunpy/coordinates/utils.py

```diff
@@ -265,3 +265,69 @@ def coordinates(self, points=None):
                         obstime=self.obstime,
                         observer=self.observer,
                         frame=Heliocentric).transform_to(self.start_frame)
+
+
+def get_rectangle_coordinates(bottom_left, *, top_right = None, width: u.deg = None, height: u.deg = None):
+    """
+    Specify a rectangular region of interest in spherical world coordinates.
+
+    Parameters
+    ----------
+    bottom_left : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The bottom_left coordinate of the rectangle. If a `SkyCoord` it can
+        have shape ``(2,)`` and also define ``top_right``.
+
+    top_right : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The top_right coordinate of the rectangle. Can only be omitted if
+        ``bottom_left`` has shape ``(2,)``.
+
+    width : `~astropy.units.Quantity`
+        The width of the rectangle.
+        Must be omitted if ``bottom_left`` has shape ``(2,)`` or ``top_right`` is provided.
+
+    height : `~astropy.units.Quantity`
+        The height of the rectangle.
+        Must be omitted if ``bottom_left`` has shape ``(2,)`` or ``top_right`` is provided.
+
+    Returns
+    -------
+    bottom_left : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The bottom_left coordinate of the rectangular region of interest in spherical world coordinates.
+
+    top_right : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`    
+        The top_right coordinate of the rectangular region of interest in spherical world coordinates.
+    """
+    if bottom_left.shape == (2,) and any((width, height, top_right)):
+        raise ValueError("Invalid input, if bottom_left.shape == (2,) \
+                          other parameters should not be passed.")
+
+    if all((width, height, top_right)):
+        raise ValueError("Invalid input, width, height and top_right \
+                          parameters should not be passed simultaneously.")
+
+    if top_right is None and bottom_left.shape != (2,) and not all((width, height)):
+        raise ValueError("Invalid input, either bottom_left and top_right \
+                         or bottom_left and height and width should be provided.")
+
+    if (not isinstance(bottom_left, (SkyCoord, BaseCoordinateFrame)) and 
+        not isinstance(top_right, (None, SkyCoord, BaseCoordinateFrame))):
+        raise TypeError("bottom_left and top_right must be of type SkyCoord or BaseCoordinateFrame")
+
+    elif bottom_left.shape == (2,):
+        top_right = bottom_left[1]
+        bottom_left = bottom_left[0]
+
+    elif top_right is not None:
+        top_right = top_right.transform_to(bottom_left.frame)        
+        width = top_right.spherical.lon - bottom_left.spherical.lon
```

Comment:
This pure subtraction doesn't work as you'd like across a longitudinal wrap.  For example, consider if a user wants to specify a 2-degree-wide-in-longitude rectangle that goes from 359 degrees to 1 degrees when wrapping at 360 degrees, or +179 degrees to -179 degrees when wrapping at 180 degrees.

Since the intent appears to be to error on rectangles than span more than 180 degrees, one way to solve these situations is to cast the difference as a `Longitude` with a wrap angle of 180 degrees.

Reply:
> This pure subtraction doesn't work as you'd like across a longitudinal wrap. For example, consider if a user wants to specify a 2-degree-wide-in-longitude rectangle that goes from 359 degrees to 1 degrees when wrapping at 360 degrees, or +179 degrees to -179 degrees when wrapping at 180 degrees.
> 
> Since the intent appears to be to error on rectangles than span more than 180 degrees, one way to solve these situations is to cast the difference as a `Longitude` with a wrap angle of 180 degrees.

In accordance with your comment above about `top_right` being "right" and "above" `bottom_left` I've removed this section all together.